pallets/api/src/messaging/benchmarking.rs

```diff
@@ -236,37 +270,57 @@ mod messaging_benchmarks {
 		}
 	}
 
+
+	/// x: Maximum context len: bound to u16::MAX T::MaxContextLen
+	/// y: Maximum length of keys (inner) len: bound to u16::MAX T::MaxKeyLen
+	/// z: Maximum amount of keys (outer) len: bound to u16::MAX T::MaxKeys
 	#[benchmark]
-	fn ismp_get() {
-		let commitment = H256::repeat_byte(2u8);
+	fn ismp_get(x: Linear<0, {u16::MAX.into()}>, y: Linear<0, {u16::MAX.into()}>, z: Linear<0, {u16::MAX.into()}>) {
 		let origin: T::AccountId = account("alice", 0, SEED);
 		let message_id = [1; 32];
+		let inner_keys: BoundedVec<u8, T::MaxKeyLen> = vec![1u8].repeat(y as usize).try_into().unwrap();
+		let mut outer_keys = vec![];
+		for k in (0..z) {
+			outer_keys.push(inner_keys.clone())
+		}
+
 		let get = crate::messaging::ismp::Get::<T> {
 			dest: 0,
 			height: 0,
 			timeout: 0,
-			context: bounded_vec!(),
-			keys: bounded_vec!(),
+			context: vec![1u8].repeat(x as usize).try_into().unwrap(),
+			keys: outer_keys.try_into().unwrap(),
 		};
 
+		pallet_balances::Pallet::<T>::make_free_balance_be(&origin, u32::MAX.into());
+
 		#[extrinsic_call]
-		Pallet::<T>::ismp_get(RawOrigin::Signed(origin.clone()), message_id, get, One::one(), None);
+		Pallet::<T>::ismp_get(RawOrigin::Signed(origin.clone()), message_id, get, None);
 	}
 
+	/// x: Maximun byte len of outgoing data.
 	#[benchmark]
-	fn ismp_post() {
-		let commitment = H256::repeat_byte(2u8);
+	fn ismp_post(x: Linear<0, {u16::MAX.into()}>) {
```

Comment:
similarly, I feel this might be better if you use the bound defined in the config (it also means the generated weights update depending on the specific runtime config)